core/io/resource_loader.cpp

```diff
@@ -685,6 +685,221 @@ float ResourceLoader::_dependency_get_progress(const String &p_path) {
 	}
 }
 
+PackedStringArray ResourceLoader::get_cached_paths(const bool &p_with_godot_type, const String &p_separator) {
+	PackedStringArray paths = PackedStringArray();
+
+	if (p_with_godot_type) {
+		for (const auto &elem : ResourceCache::resources) {
+			GDType type = elem.value->get_gdtype();
+			paths.push_back("" + type.get_name() + p_separator + elem.key);
+		}
+	} else {
+		for (const auto &elem : ResourceCache::resources) {
+			paths.push_back(elem.key);
+		}
+	}
+	return paths;
+}
```

Comment:
```suggestion
}

```
For all these, leave an empty line between methods